### PR TITLE
Improve documentation for nextcallee

### PR DESCRIPTION
**Pull request by wayland**
## The problem

nextcallee was not explained in terms of a) whether it needed parameters, and b) whether it ever returned.  

## Solution provided

- Answered both those questions
- Provided a handy table of the call/next same/with tools

### CODE REVIEW

File: doc/Language/functions.rakudoc [hunk 1]
```diff
@@ -1018,6 +1018,28 @@ dispatch, or it could be the inner routine from a C<wrap>.
...
+    samewith        | Yes     | Specify    | self
+    nextcallee's rv | Yes     | Specify    | preset
+
+=end table
+
+From the above we learn:
+
+=item C<with> always means that the parameters need to be specified
+=item C<same> at word end means the parameters of the referee will automatically be reused
+=item C<next> in an 8-letter word means that it won't return
```
Comment (coke) [inline doc/Language/functions.rakudoc:1041]: This is clever and true but not helpful. Can you describe this combination in a way that helps teach the user the difference from the others?

File: doc/Language/functions.rakudoc [hunk 2]
```diff
@@ -1018,6 +1018,28 @@ dispatch, or it could be the inner routine from a C<wrap>.
 Fortunately, we have a series of re-dispatching tools that help us to make
 it easy.
 
+=head2 The Next Tools
+
+=begin table
+
+                    | Returns | Parameters | Referee
```
Comment (coke) [inline doc/Language/functions.rakudoc:1025]: Referee as in reciprocal of referer?

File: doc/Language/functions.rakudoc [hunk 3]
```diff
@@ -1018,6 +1018,28 @@ dispatch, or it could be the inner routine from a C<wrap>.
...
+    nextcallee      | Yes     | None       | next
+    samewith        | Yes     | Specify    | self
+    nextcallee's rv | Yes     | Specify    | preset
+
+=end table
+
+From the above we learn:
+
+=item C<with> always means that the parameters need to be specified
+=item C<same> at word end means the parameters of the referee will automatically be reused
```
Comment (coke) [inline doc/Language/functions.rakudoc:1040]: can we use the same languange to describe both of these? (pease consider "at the end of the name")

File: doc/Language/functions.rakudoc (hunk 1 above)
Comment (wayland) [reply doc/Language/functions.rakudoc]: I've rewritten some of the text.  My goal in these bullet points was to point out that the same word is used with different meanings.  While "with" always has the same meaning, "same" has two different meanings, and "next" has two different meanings.  I'm highlighting some gotchas.

File: doc/Language/functions.rakudoc (hunk 2 above)
Comment (wayland) [reply doc/Language/functions.rakudoc]: Yes; thanks to your comment, I wrote some text to explain the table.  Hopefully it helps.

File: doc/Language/functions.rakudoc (hunk 3 above)
Comment (wayland) [reply doc/Language/functions.rakudoc]: I've rephrased this section.  However, I want to highlight that what I'm doing here is highlighting gotchas.  I'm not sure using the same language would support that goal, but am happy to take suggestions along that line.

File: doc/Language/functions.rakudoc [hunk 4]
```diff
@@ -1018,6 +1018,55 @@ dispatch, or it could be the inner routine from a C<wrap>.
 Fortunately, we have a series of re-dispatching tools that help us to make
 it easy.
 
+=head2 The Next Tools
+
+The table below represents the next tools in relation to three attributes:
+
+=item1 B<Returns:> "Yes" means that it returns, and the code will continue from
+that point.  "No" means that the it never returns to this function -- it's as
```
Comment (coke) [inline doc/Language/functions.rakudoc:1026]: s/the//

File: doc/Language/functions.rakudoc [hunk 5]
```diff
@@ -1018,6 +1018,55 @@ dispatch, or it could be the inner routine from a C<wrap>.
...
+
+The table below represents the next tools in relation to three attributes:
+
+=item1 B<Returns:> "Yes" means that it returns, and the code will continue from
+that point.  "No" means that the it never returns to this function -- it's as 
+though the call has a built-in "return".  
+
+=item1 B<Parameters:> Specifically, whether the parameters from the current
+function are reused in the call, or whether you need to specify some 
+parameters.  In the case of "nextcallee", there are no parameters needed,
```
Comment (coke) [inline doc/Language/functions.rakudoc:1031]: `C<nextcallee>`

File: doc/Language/functions.rakudoc [hunk 6]
```diff
@@ -1018,6 +1018,55 @@ dispatch, or it could be the inner routine from a C<wrap>.
...
+that point.  "No" means that the it never returns to this function -- it's as 
+though the call has a built-in "return".  
+
+=item1 B<Parameters:> Specifically, whether the parameters from the current
+function are reused in the call, or whether you need to specify some 
+parameters.  In the case of "nextcallee", there are no parameters needed, 
+because they're instead passed to the return value when it's called.  
+
+=item1 B<Referee:> This is the method that will be called by the tool (the 
+reciprocal of "referer".  It can be one of:
```
Comment (coke) [inline doc/Language/functions.rakudoc:1035]: Need closing )

File: doc/Language/functions.rakudoc [hunk 7]
```diff
@@ -1018,6 +1018,55 @@ dispatch, or it could be the inner routine from a C<wrap>.
...
+
+=item2 B<next:> The next item in the chain
+
+=item2 B<self:> It will call itself again, as in recursion
+
+=item2 B<preset:> It will call the item that was saved by "nextcallee"
```
Comment (coke) [inline doc/Language/functions.rakudoc:1041]: `C<nextcallee>`

File: doc/Language/functions.rakudoc [hunk 8]
```diff
@@ -1018,6 +1018,55 @@ dispatch, or it could be the inner routine from a C<wrap>.
...
+
+                    | Returns | Parameters | Referee*
+    ================+=========+============+========
+    nextsame        | No      | Reuse      | next
+    nextwith        | No      | Specify    | next
+    callsame        | Yes     | Reuse      | next
+    callwith        | Yes     | Specify    | next
+    nextcallee      | Yes     | None       | next
+    samewith        | Yes     | Specify    | self
+    nextcallee's rv | Yes     | Specify    | preset
```
Comment (coke) [inline doc/Language/functions.rakudoc:1053]: What's the rv?

File: doc/Language/functions.rakudoc [hunk 9]
```diff
@@ -1018,6 +1018,55 @@ dispatch, or it could be the inner routine from a C<wrap>.
...
+=item C<with> always means that the parameters need to be specified -- this 
+one is consistent.  
+
+=item C<same> at word end means the parameters of the currently running 
+function will automatically be reused when calling the referee.  Note that 
+C<samewith>, despite containing "same", doesn't mean that the parameters will 
+be reused.  That's because the "same" here refers to the Referee -- the 
+function is calling the same method as itself, not the next in the call stack.  
+
+=item C<next> means that it won't return, except for nextcallee, which does return
```
Comment (coke) [inline doc/Language/functions.rakudoc:1068]: `C<nextcallee>`

**Review by coke — COMMENTED** [review]
Looks good. Just some minor nits on this review.

File: doc/Language/functions.rakudoc (hunk 4 above)
Comment (wayland) [reply doc/Language/functions.rakudoc]: Done

File: doc/Language/functions.rakudoc (hunk 6 above)
Comment (wayland) [reply doc/Language/functions.rakudoc]: Done

File: doc/Language/functions.rakudoc (hunk 8 above)
Comment (wayland) [reply doc/Language/functions.rakudoc]: Return Value.  I've added a comment below the table that will hopefully explain.

File: doc/Language/functions.rakudoc (hunk 9 above)
Comment (wayland) [reply doc/Language/functions.rakudoc]: Done.

File: doc/Language/functions.rakudoc (hunk 7 above)
Comment (wayland) [reply doc/Language/functions.rakudoc]: Done

File: doc/Language/functions.rakudoc (hunk 5 above)
Comment (wayland) [reply doc/Language/functions.rakudoc]: Done

Comment (wayland) [reply doc/Language/functions.rakudoc]: @raiph I've just done another PR that should address (ie. implement) your comment as well.

**Review by coke — APPROVED** [review]
Thanks!

**Pull request merged**